### PR TITLE
curl_fopen optimization

### PR DESCRIPTION
**Pull request by SaltyMilk**

### CODE REVIEW

**Comment by gvanem** [comment]
> *No description provided.*

WTF? A PR should describe the motivation behind a change. And so what's the *optimization* here?

File: lib/fopen.c [hunk 1]
```diff
@@ -56,13 +56,13 @@ CURLcode Curl_fopen(struct Curl_easy *data, const char *filename,
...
-  if(stat(filename, &sb) == -1 || !S_ISREG(sb.st_mode)) {
-    /* a non-regular file, fallback to direct fopen() */
-    *fh = fopen(filename, FOPEN_WRITETEXT);
-    if(*fh)
-      return CURLE_OK;
+  *fh = fopen(filename, FOPEN_WRITETEXT);
+  if (!*fh)
     goto fail;
-  }
+  if (fstat(fileno(*fh), &sb) == -1 || !S_ISREG(sb.st_mode))
```
Comment (bagder) [inline lib/fopen.c:62]: `make checksrc` will tell you that we don't use spaces after `if`...

Comment (SaltyMilk) [reply lib/fopen.c]: My apologies for the oversight, fixed in next commit.

**Comment by SaltyMilk** [comment]
Last commit notes:
- make checksrc ✅ 
- Added 0600 on the fopen to silence code-scanning warning.

**Comment by bagder** [comment]
> Added 0600 on the fopen to silence code-scanning warning.

That's not how `fopen()` works (it would require the use of `open()`). The warning is wrong.

**Comment by SaltyMilk** [comment]
Sorry about that, another oversight in rush. I believe this should be good to go now. 
I'll be more careful for potential future PRs, sorry !

**Review by bagder — APPROVED** [review]

**Comment by bagder** [comment]
Thanks!

**Comment by wangdudu321123** [comment]
[【高危漏洞】CVE-2023-32001 url 存在 fopen 竞争条件漏洞 ](https://mp.weixin.qq.com/s?__biz=Mzg4MDg5NzAxMQ==&tempkey=MTIyOF9abEF0d0VwKzFRbTY0Z21CNW5MVHBtNS1YanJsUjhVZWNMOERjR0Q3cGVLak5nVExSM205bkthVUp0Q1ZyRDgyaU9qNGFfRTh5Q0gzTldxMEphdXdNVHY2ZDJJcUwwTFhSYlJwWWExSWVpLTVsNWRIbl9mNFgxc2lMNy1JM1ZmNzJraUVsczc1aE1OZElKT3pNS2RVNmJnakpsMzhFR0dSYlViSGZnfn4%3D&chksm=cf6f7a27f818f331f676cbfb1a1df9b08eb23ac38a32b721b771b1f5cba11ad64db42f9a3a29&token=1293783321&lang=zh_CN#rd)

**Pull request closed without merging**